Dscs 93

snezanam opened the pull request:
@scoen, please have a look, it has changes to include copies for the collection for alias report and expiry report query.

AaronCMuller commented on pom.xml:
```diff
@@ -3,7 +3,7 @@
   <groupId>au.gov.nla</groupId>
   <artifactId>amberdb</artifactId>
-  <version>1.2.2-SNAPSHOT</version>
+  <version>1.2.3-SNAPSHOT</version>
```
This is not actually required. Snapshots are meant to be overridden.

AaronCMuller commented on src/amberdb/AmberSession.java:
```diff
@@ -867,6 +867,24 @@ private void increment(Map<String, Integer> countMap, String key) {
+        ObjectsWithPropertyInCollectionQuery avq = new ObjectsWithPropertyInCollectionQuery(getAmberGraph());
+        List<Vertex> results = avq.getExpiryReport(expiryYear, collection);
+        List<Work> works = new ArrayList<Work>();
+        System.out.println("SIZE=" + results.size());
```
Either make this a log#info statement, or remove please.

terenceingram replied:
I think remove the line.

AaronCMuller commented on src/amberdb/query/ObjectsWithPropertyInCollectionQuery.java:
```diff
@@ -67,4 +65,54 @@ public ObjectsWithPropertyInCollectionQuery(AmberGraph graph) {
     }
+    
+    
+    public List<Vertex> getExpiryReport(String expiryYear, String collectionName) {
```
@scoen Is it possible for a list of works to be returned here?

AaronCMuller commented on src/amberdb/query/ObjectsWithPropertyInCollectionQuery.java:
```diff
@@ -67,4 +65,54 @@ public ObjectsWithPropertyInCollectionQuery(AmberGraph graph) {
+            q.execute();
+            h.commit();
+
+            // and reap the rewards
```
Please rewrite this comment to be less informal, and possibly more informative?

AaronCMuller commented:
Other than my comments, the rest looks good.

snezanam commented:
@scoen  @AaronCMuller please have another look, I have renamed the Query class and its methods and added a unit test. To generate works from vertices we need framed graph which is  available in amber session, in the query class we have AmberGraph @scoen can we easily do this?

scoen commented on src/amberdb/AmberSession.java:
```diff
@@ -867,6 +868,24 @@ private void increment(Map<String, Integer> countMap, String key) {
+
+        ObjectsWithPropertyReportQuery avq = new ObjectsWithPropertyReportQuery(getAmberGraph());
+        String tableDrop = getAmberGraph().getTempTableDrop();
+        List<Vertex> results = avq.generateExpiryReport(expiryYear, collection, tableDrop);
```
Would you mind doing the h2 check (tabledrop) inside the query itself ? Like in generateDuplicateAlias report, call graph.getTempTableDrop(). I'd prefer this code 'carbuncle' be isolated to the query class if possible. Sorry about needing to do that.

snezanam replied:
That sounds good.
